### PR TITLE
Process Consensus Messages w/o Running Duty

### PR DESCRIPTION
**Pull request by amir-blox**
This PR enables sync messages to propagate down the pipeline. Including the following changes:

- allowed consensus messages to be processed even if there is no running duty
- upon decided: set state.round only for relevant instances

### CODE REVIEW

File: qbft/decided.go [hunk 1]
```diff
@@ -34,10 +34,10 @@ func (c *Controller) UponDecided(msg *SignedMessage) (*SignedMessage, error) {
 	// Mark current instance decided
 	if inst := c.InstanceForHeight(c.Height); inst != nil && !inst.State.Decided {
 		inst.State.Decided = true
-		if msg.Message.Round > inst.State.Round {
-			inst.State.Round = msg.Message.Round
-		}
 		if c.Height == msg.Message.Height {
+			if msg.Message.Round > inst.State.Round {
```
Comment (amir-blox) [inline qbft/decided.go:38]: remove if statement

Comment (alonmuroch) [inline qbft/decided.go:38]: we can remove if, @amir-blox please add spec tests for cases that round < state round, round == state round and round > state round

File: ssv/runner.go [hunk 2]
```diff
@@ -93,15 +93,19 @@ func (b *BaseRunner) basePreConsensusMsgProcessing(runner Runner, signedMsg *Sig
 }
 
 func (b *BaseRunner) baseConsensusMsgProcessing(runner Runner, msg *qbft.SignedMessage) (decided bool, decidedValue *types.ConsensusData, err error) {
-	if err := b.validateConsensusMsg(msg); err != nil {
-		return false, nil, errors.Wrap(err, "invalid consensus message")
+	prevDecided := false
```
Comment (alonmuroch) [inline ssv/runner.go:96]: add spec tests (has running and no running)

File: ssv/runner.go [hunk 3]
```diff
@@ -93,15 +93,19 @@ func (b *BaseRunner) basePreConsensusMsgProcessing(runner Runner, signedMsg *Sig
...
+	if b.HasRunningDuty() && b.State != nil && b.State.RunningInstance != nil {
+		prevDecided, _ = b.State.RunningInstance.IsDecided()
 	}
 
-	prevDecided, _ := b.State.RunningInstance.IsDecided()
-
 	decidedMsg, err := b.QBFTController.ProcessMsg(msg)
 	if err != nil {
-		return false, nil, errors.Wrap(err, "failed to process consensus msg")
+		return false, nil, err
```
Comment (alonmuroch) [inline ssv/runner.go:103]: why removed wrapping?

**Comment by amir-blox** [comment]
Review Items:
- add previous decided test

**Review by alonmuroch — CHANGES_REQUESTED** [review]

File: ssv/runner.go (hunk 2 above)
Comment (lior-blox) [reply ssv/runner.go:96]: Already exist

File: ssv/runner.go (hunk 3 above)
Comment (amir-blox) [reply ssv/runner.go:103]: We wrap the err with this string also in the runner level (e.g. attester)

**Review by lior-blox — APPROVED** [review]

**Comment by amir-blox** [comment]
https://github.com/bloxapp/ssv-spec/pull/94/commits/130062072d954b5b9396f1b2cb102d54ab4a369b
Was reverted as it caused errors in other tests, we should do it properly (errors alignment cross project)

File: qbft/decided.go (hunk 1 above)
Comment (amir-blox) [reply qbft/decided.go]: The tests already exists:
`decided.CurrentInstance()`
`decided.CurrentInstanceFutureRound()`
`decided.CurrentInstancePastRound()` - fixed ctrl root as now we use the right round

**Review by nivBlox — APPROVED** [review]

**Comment by alonmuroch** [comment]
@amir-blox ok by me, please rebase and squash

**Review by alonmuroch — APPROVED** [review]
approved

**Pull request merged**